### PR TITLE
Better descriptors.

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context / Related issue

good descriptors = beautiful plots = cool for publishing articles.

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/benchmark/experiments.py [hunk 1]
```diff
@@ -40,25 +40,19 @@ def mltuning(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
...
             for optim in optims:
-                for function in functions:
-                    xp = Experiment(function, optim, num_workers=num_workers,
-                                    budget=budget, seed=next(seedg))
-                    if not xp.is_incoherent:
-                         yield xp
+                for dimension in [1, 2, 3]:
+                    for regressor in ["mlp", "decision_tree", "decision_tree_depth"]:
+                        function = MLTuning(regressor, dimension)
+                        function.add_descriptors(regressor=regressor, dimension=dimension)
```
Comment (jrapin) [inline nevergrad/benchmark/experiments.py:51]: the descriptors should be directly implemented *inside* the function

File: nevergrad/benchmark/experiments.py [hunk 2]
```diff
@@ -40,25 +40,19 @@ def mltuning(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
...
         for num_workers in [1, 10, 50, 100]:
             for optim in optims:
-                for function in functions:
-                    xp = Experiment(function, optim, num_workers=num_workers,
-                                    budget=budget, seed=next(seedg))
-                    if not xp.is_incoherent:
-                         yield xp
+                for dimension in [1, 2, 3]:
+                    for regressor in ["mlp", "decision_tree", "decision_tree_depth"]:
+                        function = MLTuning(regressor, dimension)
```
Comment (jrapin) [inline nevergrad/benchmark/experiments.py:50]: please reorganize the for loops so as to instantiate the function as soon as possible, that will save a lot of time in the tests. That means basically moving budget, numworkers etc for loops after dimension and regressor

File: nevergrad/functions/ml/mlfunctionlib.py [hunk 3]
```diff
@@ -28,7 +29,7 @@ class MLTuning(ExperimentFunction):
     # Example of ML problem.
     def _ml_parametrization(self,
                             depth: int,  # Parameters for regression trees.
-                            dimension: int,
+                            data_dimension: int,
```
Comment (jrapin) [inline nevergrad/functions/ml/mlfunctionlib.py:32]: why not dimension? to differentiate from the dimension of the optimization?
maybe data_size is shorter and makes the point that it's different

File: nevergrad/functions/ml/mlfunctionlib.py [hunk 4]
```diff
@@ -148,11 +152,23 @@ def __init__(self, regressor: str, dimension: int):
...
+            super().__init__(partial(self._ml_parametrization, data_dimension=data_dimension, noise_free=False,
                                      regressor="mlp", depth=-3, criterion="no", min_samples_split=0.1), parametrization)
-            self.evaluation_function = partial(self._ml_parametrization, dimension=dimension,  # type: ignore
+            self.evaluation_function = partial(self._ml_parametrization, data_dimension=data_dimension,  # type: ignore
                                                regressor="mlp", noise_free=True, 
                                                depth=-3, criterion="no", min_samples_split=0.1)
         else:
             assert False, f"Problem type {regressor} undefined!"
-        self.register_initialization(regressor=regressor, dimension=dimension)
+        self.register_initialization(regressor=regressor, data_dimension=data_dimension)
```
Comment (jrapin) [inline nevergrad/functions/ml/mlfunctionlib.py:162]: the descriptors are probably missing here

File: nevergrad/functions/ml/mlfunctionlib.py [hunk 5]
```diff
@@ -148,11 +152,23 @@ def __init__(self, regressor: str, dimension: int):
...
+
+
+    def copy(self):
```
Comment (jrapin) [inline nevergrad/functions/ml/mlfunctionlib.py:165]: that should not be necessary if the parametrization is created during the init, and all the init parameters are well registered. Why did you add it?

File: nevergrad/functions/ml/mlfunctionlib.py (hunk 3 above)
Comment (teytaud) [reply nevergrad/functions/ml/mlfunctionlib.py:32]: Yes, for that reason.
data_size is ambiguous to me, whereas data_dimension looks good enough.

File: nevergrad/functions/ml/mlfunctionlib.py (hunk 4 above)
Comment (teytaud) [reply nevergrad/functions/ml/mlfunctionlib.py:162]: done.

File: nevergrad/functions/ml/mlfunctionlib.py (hunk 5 above)
Comment (teytaud) [reply nevergrad/functions/ml/mlfunctionlib.py:165]: Ok!

File: nevergrad/benchmark/experiments.py (hunk 1 above)
Comment (teytaud) [reply nevergrad/benchmark/experiments.py]: Done.

File: nevergrad/benchmark/experiments.py (hunk 2 above)
Comment (teytaud) [reply nevergrad/benchmark/experiments.py]: Done.
